src/proxyvisor.coffee

```diff
@@ -0,0 +1,268 @@
+device = require './device'
+bodyParser = require 'body-parser'
+request = Promise.promisifyAll require 'request'
+config = require './config'
+PUBNUB = require 'pubnub'
+execAsync = Promise.promisify(require('child_process').exec)
+
+pubnub = PUBNUB.init(config.pubnub)
+
+getAssetsPath = (image) ->
+	docker.imageRootDir(image)
+	.then (rootDir) ->
+		return rootDir + '/assets'
+
+exports.router = router = express.Router()
+router.use(bodyParser())
+
+router.get '/v1/devices', (req, res) ->
+	knex('dependentDevice').select()
+	.then (devices) ->
+		res.json(devices)
+	.catch (err) ->
+		res.status(503).send(err?.message or err or 'Unknown error')
+
+router.post '/v1/devices', (req, res) ->
+	Promise.join(
+		utils.getConfig('apiKey')
+		utils.getConfig('userId')
+		device.getID()
+		deviceRegister.generateUUID()
+		randomHexString.generate()
+		(apiKey, userId, deviceId, uuid, logsChannel) ->
+			d =
+				user: userId
+				application: req.body.applicationId
+				uuid: uuid
+				device_type: 'edge'
+				device: deviceId
+				registered_at: Math.floor(Date.now() / 1000)
+				logs_channel: logsChannel
+				status: 'Provisioned'
+			resinApi.post
+				resource: 'device'
+				body: d
+				customOptions:
+					apikey: apiKey
+			.then (dev) ->
+				deviceForDB = {
+					uuid: uuid
+					appId: d.application
+					device_type: d.device_type
+					deviceId: dev.id
+					name: dev.name
+					status: d.status
+					logs_channel: d.logs_channel
+				}
+				knex('dependentDevice').insert(deviceForDB)
+				.then ->
+					res.status(202).send(dev)
+	)
+	.catch (err) ->
+		console.error('Error on GET /v1/devices:', err, err.stack)
+		res.status(503).send(err?.message or err or 'Unknown error')
+
+router.get '/v1/devices/:uuid', (req, res) ->
+	uuid = req.params.uuid
+	knex('dependentDevice').select().where({ uuid })
+	.then ([ device ]) ->
+		return res.status(404).send('Device not found') if !device?
+		res.json(device)
+	.catch (err) ->
+		console.error('Error on GET /v1/devices/:uuid:', err, err.stack)
+		res.status(503).send(err?.message or err or 'Unknown error')
+
+router.post '/v1/devices/:uuid/logs', (req, res) ->
+	uuid = req.params.uuid
+	m = {
+		message: req.body.message
+		timestamp: req.body.timestamp or Date.now()
+	}
+	m.isSystem = req.body.isSystem if req.body.isSystem?
+
+	knex('dependentDevice').select().where({ uuid })
+	.then ([ device ]) ->
+		return res.status(404).send('Device not found') if !device?
+		pubnub.publish({ channel: "device-#{device.logs_channel}-logs", message: m })
+		res.status(202).send('OK')
+	.catch (err) ->
+		console.error('Error on PUT /v1/devices/:uuid/logs:', err, err.stack)
+		res.status(503).send(err?.message or err or 'Unknown error')
+
+router.put '/v1/devices/:uuid', (req, res) ->
+	uuid = req.params.uuid
+	status = req.body.status
+	is_online = req.body.is_online
+	if !is_online? and !status?
+		res.status(400).send('status or is_online must be set')
+		return
+	if _.isNull(status) or _.isNull(is_online)
```

Comment:
I think you could change this to:
```coffee
if !_.isBoolean(is_online)
	return res.status(400).send('is_online must be a boolean')
if _.isEmpty(status)
	return res.status(400).send('status must be a non-empty string')
```
To reduce the complexity of the checks a bit